airflow/cli/commands/scheduler_command.py

```diff
@@ -60,4 +69,17 @@ def scheduler(args):
         signal.signal(signal.SIGINT, sigint_handler)
         signal.signal(signal.SIGTERM, sigint_handler)
         signal.signal(signal.SIGQUIT, sigquit_handler)
+        if is_local_executor:
+            sub_proc = _serve_logs(skip_serve_logs)
```

Comment:
Can you check that when sending a SIGINT (i.e. ctrl-c) that the log server shuts down.